providers/tests/system/google/cloud/cloud_sql/example_cloud_sql_query_iam.py

```diff
@@ -0,0 +1,491 @@
+            hook = CloudSQLHook(api_version="v1", gcp_conn_id=connection_id)
+            certificate_name = f"test_cert_{''.join(random.choice(string.ascii_letters) for _ in range(8))}"
+            response = hook.create_ssl_certificate(
+                instance=instance,
+                body={"common_name": certificate_name},
+                project_id=PROJECT_ID,
+            )
+            return response
+
+        create_ssl_certificate_task = create_ssl_certificate(
+            instance=cloud_sql_instance_name, connection_id=create_connection_task
+        )
+
+        @task(task_id=f"save_ssl_cert_locally_{database_type}")
+        def save_ssl_cert_locally(ssl_cert: dict[str, Any], db_type: str) -> dict[str, str]:
+            folder = SSL_COMPOSER_PATH_PREFIX if run_in_composer() else SSL_LOCAL_PATH_PREFIX
+            folder += f"/certs/{db_type}/{ssl_cert['operation']['name']}"
+            if not os.path.exists(folder):
+                os.makedirs(folder)
+            _ssl_root_cert_path = f"{folder}/sslrootcert.pem"
+            _ssl_cert_path = f"{folder}/sslcert.pem"
+            _ssl_key_path = f"{folder}/sslkey.pem"
+            with open(_ssl_root_cert_path, "w") as ssl_root_cert_file:
+                ssl_root_cert_file.write(ssl_cert["serverCaCert"]["cert"])
+            with open(_ssl_cert_path, "w") as ssl_cert_file:
+                ssl_cert_file.write(ssl_cert["clientCert"]["certInfo"]["cert"])
+            with open(_ssl_key_path, "w") as ssl_key_file:
+                ssl_key_file.write(ssl_cert["clientCert"]["certPrivateKey"])
+            return {
+                "sslrootcert": _ssl_root_cert_path,
+                "sslcert": _ssl_cert_path,
+                "sslkey": _ssl_key_path,
+            }
+
+        save_ssl_cert_locally_task = save_ssl_cert_locally(
+            ssl_cert=create_ssl_certificate_task, db_type=database_type
+        )
+
+        @task(task_id=f"save_ssl_cert_to_secret_manager_{database_type}")
+        def save_ssl_cert_to_secret_manager(ssl_cert: dict[str, Any], db_type: str) -> str:
+            hook = GoogleCloudSecretManagerHook()
+            payload = {
+                "sslrootcert": ssl_cert["serverCaCert"]["cert"],
+                "sslcert": ssl_cert["clientCert"]["certInfo"]["cert"],
+                "sslkey": ssl_cert["clientCert"]["certPrivateKey"],
+            }
+            _secret_id = f"secret_{DAG_ID}_{ENV_ID}_{db_type}"
+
+            if not hook.secret_exists(project_id=PROJECT_ID, secret_id=_secret_id):
+                hook.create_secret(
+                    secret_id=_secret_id,
+                    project_id=PROJECT_ID,
+                )
+
+            hook.add_secret_version(
+                project_id=PROJECT_ID,
+                secret_id=_secret_id,
+                secret_payload=dict(data=base64.b64encode(json.dumps(payload).encode("ascii"))),
+            )
+
+            return _secret_id
```

Comment:
What is the point of creating a secret and later deleting it?  Am I missing something and it got used somewhere under the hood?

Reply:
You are right this piece of code is not needed. I have removed it.